connectors/crowdstrike2tone/crowdstrike/transform.py

```diff
@@ -127,27 +148,63 @@ def transform_asset(self, asset: dict[str, Any]) -> dict[str, Any]:
-        #    'object_type': 'cve-finding',
-        #    'asset_id': finding['machineId'],
-        #    'id': str(uuid3(self.NAMESPACE, finding['id'])),
-        #    'cve': {'cves': [finding['cveId']]},
-        #    'exposure': {'severity': {'level': str(finding['severity'].upper())}},
-        #    'observations': {
-        #        'software': [
-        #            {
-        #                'product': {
-        #                    'product_name': trunc(str(finding['productName']), 32),
-        #                    'vendor_name': trunc(str(finding['productVendor']), 32),
-        #                    'version': trunc(str(finding['productVersion']), 32),
-        #                },
-        #            },
-        #        ],
-        #    },
-        # }
-
-
-#
+        # Map CrowdStrike finding status to T1 finding state
+        status_switch = {
+            'open': 'ACTIVE',
+            'reopen': 'REOPENED',
+        }
+
+        # Extract the relevant information from the finding
+        observations = {}
+        if finding.get('apps'):
+            observations['software'] = []
+            for app in finding['apps']:
+                observations['software'].append(
+                    {
+                        'product': {
+                            # Normalize the product name and vendor to 32 characters
+                            'product_name': trunc(
+                                str(app.get('vendor_normalized', '')), 32
+                            ),
+                            'vendor_name': trunc(
+                                str(app.get('product_name_version', '')), 32
+                            ),
+                            'version': trunc(
+                                str(app.get('product_name_normalized', '')), 32
+                            ),
+                        }
+                    }
+                )
+        CVE_ID = finding.get('cve', {}).get('id')
+        SEVERITY = finding.get('cve', {}).get('severity')
+        # Normalize the severity
+        if SEVERITY:
+            SEVERITY = SEVERITY if SEVERITY != 'UNKNOWN' else 'NONE'
+
+        # Return the transformed finding
+        return {
+            'object_type': 'cve-finding',
+            'asset_id': finding['aid'],
+            'definition_urn': f'urn:crowdstrike:{finding["vulnerability_id"]}',
+            # Use the status switch to map the CrowdStrike finding status to
+            # T1 finding state
+            'state': status_switch.get(finding.get('status', ''), 'ACTIVE'),
+            'discovery': {
+                'first_observed_at': finding.get('created_timestamp'),
+                'last_observed_on': finding.get('updated_timestamp'),
+            },
+            'id': finding['id'],
+            'cve': {'cves': [CVE_ID]} if CVE_ID else None,
+            'observations': observations if observations else None,
+            'exposure': {'severity': {'level': SEVERITY} if SEVERITY else None},
+        }
```

Comment:
uppercase variable names are generally reserved for constants.

```suggestion
        severity = finding.get('cve', {}).get('severity')
        # Normalize the severity
        if severity:
            severity = severity if severity != 'UNKNOWN' else None

        # Return the transformed finding
        return {
            'object_type': 'cve-finding',
            'asset_id': finding['aid'],
            'definition_urn': f'urn:crowdstrike:{finding["vulnerability_id"]}',
            # Use the status switch to map the CrowdStrike finding status to
            # T1 finding state
            'state': status_switch.get(finding.get('status', ''), 'ACTIVE'),
            'discovery': {
                'first_observed_at': finding.get('created_timestamp'),
                'last_observed_on': finding.get('updated_timestamp'),
            },
            'id': finding['id'],
            'cve': {'cves': [cve_id]} if finding.get('cve', {}).get('id') else None,
            'observations': observations if observations else None,
            'exposure': {'severity': {'level': severity} if severity else None},
        }
```